html5/js/OffscreenDecodeWorker.js

```diff
@@ -418,81 +158,68 @@ class WindowDecoder {
           sw,
           sh
         );
-        if (this.debug) {
-          this.paint_box(context, "brown", sx + xdelta, sy + ydelta, sw, sh);
-        }
+        this.paint_box(coding, context, sx, sy, sw, sh);
       }
-    } else if (coding == "frame") {
+    } else if (coding.startsWith("frame")) {
       let enc_width = width;
       let enc_height = height;
       const options = packet[10] || {};
       const scaled_size = options["scaled_size"];
-      if (scaled_size) {
+      if (scaled_size && (enc_width > width || enc_height > height)) {
```

Comment:
Why this new check?
We can conceivably downscale too - we generally prefer scaling before compressing but this should not be impossible.

Reply:
Did this because enc_width & enc_height where smaller then width and height, so I saw video corruption. This is a dirty fix. Need to figure out where the real issue is.

Reply:
Ah, I think this is the wrong place to use scaled_size; width & height are the width & height on screen (destination). scaled_size is only interesting when feeding the frame to VideoDecoder (however, VideoDecoder seems to figure it out itself)